backend/apiviews/author_views.py

```diff
@@ -60,3 +60,24 @@ def get_friends(self, request, pk, *args, **kwargs):
             id_List.append(list(list(i.items())[0][1].items())[0][1])
 
         return Response({"query": "friends", "Author": id_List})
+    
+    def check_friends(self, request, authorId1,authorId2, *args, **kwargs):
+    
+        id1 = protocol_removed(authorId1)
+        id2 = protocol_removed(authorId2)
+        author1 = get_object_or_404(User, fullId=id1)
+        author2 = get_object_or_404(User, fullId=id2)
+        friend_ids = Friend.objects.filter(
+            fromUser_id=author1.id).values_list('toUser_id', flat=True)
+        friends_list = []
+        for id in friend_ids:
+            friends_list.append(id)
```

Comment:
What is this function doing? It seems to be just copying items from friends_ids to friends_list so why not just use friend_ids?

Reply:
This function is being called in other API views too so we might have to change all of those as well

Reply:
I updated this. friends_id returned a <QuerySet> and now i am just checking if the ids exist in there